assets/src/vo/date-time/server-date-time.js

```diff
@@ -25,34 +25,35 @@ import moment from 'moment-timezone';
  * has a timezone string set or is using a UTC offset.
  */
 export default class ServerDateTime extends DateTime {
+
 	/**
 	 * The constructor for the ServerDateTime class
 	 *
 	 * @param {string} iso8601DateString
-	 * @param {string} locale
 	 * @param {string} timezone
+	 * @param {string} locale
 	 */
 	constructor(
 		iso8601DateString = '',
-		locale = DEFAULT_VALID_LOCALE,
 		timezone = DEFAULT_TIMEZONE_STRING,
+		locale = DEFAULT_VALID_LOCALE,
 	) {
 		// we only want to use the timezone value if the server indicates there
 		// is a a timezone string or if constructing an instance for a non UTC
 		// value timezone (HAS_TIMEZONE_STRING is just a shortcut check).
 		if (
 			HAS_TIMEZONE_STRING ||
 			( !! timezone && timezone !== 'UTC' )
-		) {
-			super( iso8601DateString, timezone, locale );
-		} else {
-			const datetime = !! iso8601DateString ?
+			) {
+				super( iso8601DateString, timezone, locale, 'ServerDateTime' );
```

Comment:
had to pass the class name to the parent class because it uses `Object.freeze` which prevents setting properties after calling super... and you JS doesn't like when you set properties before you call super, soooo.... here we are!

Reply:
JS is not for OOP, 😄